### ISSUE TITLE
`FA100` and `FA102` do not trigger with function annotations

### ISSUE THREAD

**Issue by aazuspan** [astral-sh/ruff#5574]
I'm migrating a project to `ruff` (huge fan so far, thanks maintainers!), and getting inconsistent behavior from the `flake8-future-annotations` rules, [FA100](https://beta.ruff.rs/docs/rules/future-rewritable-type-annotation/#future-rewritable-type-annotation-fa100) and [FA102](https://beta.ruff.rs/docs/rules/future-required-type-annotation/#future-required-type-annotation-fa102). They trigger as expected for variable type annotations but do not trigger for function parameter or return type annotations.

## Reproducing

`pyproject.toml` config.
```toml
[tool.ruff]
target-version = "py38"
select = ["FA"]
```

`main.py` using the example code from the `ruff` docs.
```python
from typing import List


# Doesn't trigger FA100
def foo(a: List[str]):
    ...

# Doesn't trigger FA102
def bar(a: int | None):
    ...
```

And the commands I'm running:

```bash
$ ruff --version
0.0.277

$ ruff check ./main.py -v
[2023-07-06][17:10:49][ruff_cli::commands::run][DEBUG] Identified files to lint in: 2.948576ms
[2023-07-06][17:10:49][ruff_cli::diagnostics][DEBUG] Checking: /home/az/ruff_test/main.py
[2023-07-06][17:10:49][ruff_cli::commands::run][DEBUG] Checked 1 files in: 299.315µs
```

No rules triggered.

## Exceptions

Both rules *will* trigger with variable type annotations. For example:

`main.py`
```python
from typing import List


# Does trigger FA100
def foo():
    a: List[str]

# Does trigger FA102
def bar():
    a: int | None
```

```bash
$ ruff check ./main.py -v
[2023-07-06][17:10:02][ruff_cli::commands::run][DEBUG] Identified files to lint in: 1.427045ms
[2023-07-06][17:10:02][ruff_cli::diagnostics][DEBUG] Checking: /home/az/ruff_test/main.py
[2023-07-06][17:10:02][ruff_cli::commands::run][DEBUG] Checked 1 files in: 416.495µs
main.py:19:8: FA100 Missing `from __future__ import annotations`, but uses `typing.List`
main.py:23:8: FA102 Missing `from __future__ import annotations`, but uses PEP 604 union
Found 2 errors.
```

This seems to be inconsistent with `flake8-future-annotations`, the example code in the `ruff` docs, and I believe the intended behavior. Let me know if you need any more details, thanks!

**Comment by charliermarsh** [comment]
Thanks for the clear issue. I need to look back to refresh myself on this.

**Comment by charliermarsh** [comment]
Yeah this is a bug. Thanks! I'll make sure it's fixed for the next release.

**Comment by aazuspan** [comment]
Awesome, thanks for such a quick response!
